stdlib/messaging/rabbitmq/src/main/ballerina/src/rabbitmq/channel.bal

```diff
@@ -14,88 +14,126 @@
 // specific language governing permissions and limitations
 // under the License.
 
-# Public Ballerina API - Ballerina interface to an AMQP `Channel`.
-# To provide AMQ `Channel` related functionalities.
-#
-# + connection - Reference to a Ballerina RabbitMQ `Connection`.
+import ballerinax/java;
+
+# Ballerina interface to provide AMQP `Channel` related functionality.
 public type Channel client object {
 
-    Connection? connection;
+    handle amqpChannel;
 
     # Initializes a Ballerina `Channel` object with the given `Connection` object or connection parameters.
     # Creates a `Connection` object if only the connection configuration is given.
     #
     # + connectionOrConnectionConfig - Holds a Ballerina RabbitMQ `Connection` object or the connection parameters.
     public function __init(ConnectionConfiguration|Connection connectionOrConnectionConfig) {
-        self.connection = (connectionOrConnectionConfig is Connection) ?
+        Connection connection = (connectionOrConnectionConfig is Connection) ?
                                 connectionOrConnectionConfig : new Connection(connectionOrConnectionConfig);
-        self.createChannel(self.connection);
+        self.amqpChannel = createChannel(connection.amqpConnection);
     }
 
-    # Actively declare a non-exclusive, autodelete, non-durable queue, or queue with the given configurations.
+    # Actively declare a non-exclusive, auto-delete, non-durable queue, or queue with the given configurations.
     #
     # + queueConfig - Holds the parameters required to declare a queue.
     # + return - The name of the queue if autogenerated or nil if the queue was successfully
     #               generated with the given parameters. An error is returned if an I/O error is encountered.
-    public remote function queueDeclare(QueueConfiguration? queueConfig = ())
-                                        returns string | Error? = external;
+    public remote function queueDeclare(QueueConfiguration? queueConfig = ()) returns string | Error? {
```

Comment:
Is this signature correct? In which situation this return ()?

Reply:
Yes it is correct. 
return string - If the QueueConfiguration is not given (if queueConfig is ()) server generates a queue with a random name, so that server generated name is returned. User needs this name to publish messages or to subscribe. 
return () - if the QueueConfiguration is given by the user with a queue name and creating the queue is successful.
return error - if any error occurs in any of the above situations